### PR TITLE
migrated to v2 and android x

### PR DESCRIPTION
**Pull request by abdelaziz-mahdy**

### CODE REVIEW

File: example/.flutter-plugins-dependencies [hunk 1]
```diff
@@ -0,0 +1 @@
+{"info":"This is a generated file; do not edit or check into version control.","plugins":{"ios":[],"android":[{"name":"android_path_provider","path":"D:\\\\test\\\\android_path_provider-master\\\\","dependencies":[]}],"macos":[],"linux":[],"windows":[],"web":[]},"dependencyGraph":[{"name":"android_path_provider","dependencies":[]}],"date_created":"2021-09-24 10:26:52.344691","version":"2.5.0"}
```
Comment (steven-spiel) [inline example/.flutter-plugins-dependencies:1]: May want to add this file to gitignore

File: .idea/workspace.xml [hunk 2]
```diff
@@ -1,45 +1,83 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (steven-spiel) [inline .idea/workspace.xml:1]: All of your .idea files can be added to .gitignore

**Review by steven-spiel — COMMENTED** [review]
This is great! It has a heavier footprint than I would expect from this change. Seems like the only files that actually need updating are

- android/src/main/kotlin/com/mix1009/android_path_provider/AndroidPathProviderPlugin.kt
- android/gradle.properties

and from the example:

- example/android/app/src/main/AndroidManifest.xml
- example/android/app/src/main/kotlin/com/mix1009/android_path_provider_example/MainActivity.kt
- example/android/gradle.properties

The rest of the changes can either be removed or ignored

**Comment by abdelaziz-mahdy** [comment]
> This is great! It has a heavier footprint than I would expect from this change. Seems like the only files that actually need updating are
> 
> - android/src/main/kotlin/com/mix1009/android_path_provider/AndroidPathProviderPlugin.kt
> - android/gradle.properties
> 
> and from the example:
> 
> - example/android/app/src/main/AndroidManifest.xml
> - example/android/app/src/main/kotlin/com/mix1009/android_path_provider_example/MainActivity.kt
> - example/android/gradle.properties
> 
> The rest of the changes can either be removed or ignored

Yes but the pull request didn't ignore them and couldn't force ignore them 

If you know how to ignore them,please tell me so I do it

File: .idea/workspace.xml (hunk 2 above)
Comment (abdelaziz-mahdy) [reply .idea/workspace.xml:1]: Ok will do so and make another pull request 

But it may take some time since I am not home

**Pull request merged**